src/core/utils/time/timeUtils.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  DAY_IN_HRS,
+  DAY_IN_S,
+  HOUR_IN_S,
+  MINUTE_IN_S,
+  SECOND_IN_MS
+} from "./timeConstants";
+import {CountdownResults} from "./timeTypes";
+import {CountdownItem} from "../../../countdown/Countdown";
+
+function calculateRemainingTime(target: Date): CountdownResults {
```

Comment:
I'm not 100% happy about the name of the function and the type here. Maybe this is better. I think we should always try to add documentation for the utilities we define, so let's do that as well
```suggestion
function calculateRemainingTimeBreakdown(target: Date): RemainingTimeBreakdown {
```